sbi/inference/snpe/snpe_base.py

```diff
@@ -155,10 +155,16 @@ def __call__(
 
             # Run simulations for the round.
             theta, x, prior_mask = self._run_simulations(round_, num_sims)
+            # XXX: Bypassing binary calibration kernel here. We need this to get rid of
+            # NaN simulations. We could do that using calibration kernels, but then we
+            # hack here can be added to sre and snl easily.
```

Comment:
Why should the lack of calibration kernels in SRE/SNL lead to a hack in SNPE? Is it just for uniformity?

If that's the case, I'd just add the hack to where its needed and explain 'we need this here because SRE/SNL don't have support for calibration kernels'.